### PR TITLE
Fixing the nightly job

### PR DESCRIPTION
**Pull request by daviddavis**
[noissue]

### CODE REVIEW

**Comment by daviddavis** [comment]
This is not quite right. We need to somehow split the test steps that use the matrix from the other steps (publish docs, clients, etc).

**Comment by fao89** [comment]
from L70 it needs to be another job that depends on the previous one

File: templates/github/.github/workflows/nightly.yml.j2 [hunk 1]
```diff
@@ -59,6 +57,51 @@ jobs:
         run: .github/workflows/scripts/script.sh
         shell: bash
 
+  publish:
+    runs-on: ubuntu-latest
+    needs: test
+
+    env:
+      - TEST: docs
```
[the remark this answers is not in the document]
Comment (daviddavis) [reply templates/github/.github/workflows/nightly.yml.j2:65]: It looks like some steps need `TEST=bindings` while others need `TEST=docs`. I need to either:
1. Break these steps up into jobs each with their own before_install/install/..
2. Rewrite the scripts. Maybe add a `TEST=publish` or something?

[the remark this answers is not in the document]
Comment (fao89) [reply templates/github/.github/workflows/nightly.yml.j2:65]: I don't have a strong opinion on that, I would pick the easier/faster way, or maybe move the docs to publish docs workflow?

**Review by fao89 — APPROVED** [review]
Awesome!

**Pull request merged**